lib/src/json_schema/constants.dart

```diff
@@ -158,23 +158,40 @@ class SchemaVersion implements Comparable<SchemaVersion> {
   }
 }
 
-Map getJsonSchemaDefinitionByRef(String ref) {
-  final mapping = {
-    SchemaVersion.draft4.toString(): JsonSchemaDefinitions.draft4,
-    SchemaVersion.draft6.toString(): JsonSchemaDefinitions.draft6,
-    SchemaVersion.draft7.toString(): JsonSchemaDefinitions.draft7,
-    SchemaVersion.draft2019_09.toString(): JsonSchemaDefinitions.draft2019_09,
-  };
+final _staticSchemaMapping = {
+  parseStandardizedUri(SchemaVersion.draft4.toString()): JsonSchemaDefinitions.draft4,
+  parseStandardizedUri(SchemaVersion.draft6.toString()): JsonSchemaDefinitions.draft6,
+  parseStandardizedUri(SchemaVersion.draft7.toString()): JsonSchemaDefinitions.draft7,
+  parseStandardizedUri(SchemaVersion.draft2019_09.toString()): JsonSchemaDefinitions.draft2019_09,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/meta/validation"): Draft2019Subschemas.validation,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/vocab/validation"): Draft2019Subschemas.validation,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/meta/format"): Draft2019Subschemas.format,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/vocab/format"): Draft2019Subschemas.format,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/meta/core"): Draft2019Subschemas.core,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/vocab/core"): Draft2019Subschemas.core,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/meta/metadata"): Draft2019Subschemas.metadata,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/vocab/metadata"): Draft2019Subschemas.metadata,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/meta/applicator"): Draft2019Subschemas.applicator,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/vocab/applicator"): Draft2019Subschemas.applicator,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/meta/content"): Draft2019Subschemas.content,
+  parseStandardizedUri("https://json-schema.org/draft/2019-09/vocab/content"): Draft2019Subschemas.content,
+};
 
-  if (SchemaVersion.values.map((value) => value.toString()).contains(ref)) {
-    return json.decode(mapping[ref]);
-  }
+Uri parseStandardizedUri(String s) => standardizeUri(Uri.parse(s));
+Uri standardizeUri(Uri uri) => uri?.replace(scheme: "https", fragment: null);
```

Comment:
What happens if we get a `file` scheme? Might want to conditionally replace

Reply:
I considered that and thought replacing made sense, but upon reconsideration I agree with you

Reply:
Updated